boot/bootutil/src/bootutil_public.c

```diff
@@ -528,6 +542,72 @@ boot_set_next(const struct flash_area *fa, bool active, bool confirm)
 
     return rc;
 }
+#else
+int
+boot_set_next(const struct flash_area *fa, bool active, bool confirm)
+{
+    struct boot_swap_state slot_state;
+    int rc;
+
+    if (active) {
+        /* The only way to set active slot for next boot is to confirm it,
+         * as DirectXIP will conclude that, since slot has not been confirmed
+         * last boot, it is bad and will remove it.
```

Comment:
Seems like this should be reworked, in swap using move you can have 2 applications constantly and flip between them, I don't see why this should be any different for XIP - might have an end application in one slot and e.g. a FOTA update receiver in the other, if mcuboot will always erase the slot which is not active, that seems like a bug. Outside the scope of this PR however.

Reply:
That would require rework of how we set/reset flags for DIrectXIP.
Using retention API we could avoid writing any bits to flash just to test alternative image, so this is also an option here.